openedx/core/djangoapps/content/search/views.py

```diff
@@ -29,6 +33,21 @@ def _get_meili_api_key_uid():
     return _get_meili_api_key_uid.uid
 
 
+def _get_meili_access_filter(request: Request) -> dict:
+    """
+    Return meilisearch filter based on the requesting user's permissions.
+    """
+    # Global staff can see anything, so no filters required.
+    if GlobalStaff().has_user(request.user):
+        return {}
+
+    # Everyone else is limited to the N most recent courses and libraries they can access.
+    access_ids = get_access_ids_for_request(request)[:MAX_ACCESS_IDS_IN_FILTER]
+    return {
+        "filter": f"access_id IN {access_ids}",
+    }
```

Comment:
Is it possible to separate "org level access" from "course level access", as stated in the issue? Like: `org IN ["orgA", "orgB", ...] OR access_id IN [1, 2, 3, ...]`

That way, we rely less on the `access_id` and its "limitations".

Reply:
Please see [3b07501](https://github.com/openedx/edx-platform/pull/34471/commits/3b075014b2aea8069c3900b407bd6622508a2a46)